### ISSUE TITLE
Error when using np.where() during squad tokenization 

### ISSUE THREAD

**Issue by Adawindcatcher** [huggingface/transformers#6083]
# 🐛 Bug

## Information

Model I am using (Bert, XLNet ...):Bert

Language I am using the model on (English, Chinese ...):English

The problem arises when using:
* [x] the official example scripts: squad1.1
* [ ] my own modified scripts: (give details below)

The tasks I am working on is:
* [x] an official GLUE/SQUaD task: SQUaD task

* [ ] my own task or dataset: (give details below)

## To reproduce
 I'm confused about this line:
https://github.com/huggingface/transformers/blob/896300177bf9f35feac4698370212a80a5ab6138/src/transformers/data/processors/squad.py#L230

Should this be  
```
pad_token_indices = np.where(np.array(span["input_ids"]) == tokenizer.pad_token_id)
```
Cause "p_mask: mask with 1 for token than cannot be in the answer (0 for token which can be in an answer)", and the padding positions should not be answers

```python
span["input_ids"] = [1,1,3,0,0,0]
np.where([1,1,3,0,0,0] == tokenizer.pad_token_id)
# return (array([], dtype=int64),)

np.where(np.array([1,1,3,0,0,0]) == tokenizer.pad_token_id)
# return (array([3, 4, 5]),)
```

## Expected behavior

As described above.

## Environment info

     
- `transformers` version: 7a68d401388bc68f10dfeb591709352736a6c0b6
- Platform:
- Python version: python3.6.8
- PyTorch version (GPU?):1.5.0 CPU
- Tensorflow version (GPU?):
- Using GPU in script?: NO
- Using distributed or parallel set-up in script?:NO
